plug-ins/dns_spoof/dns_spoof.c

```diff
@@ -251,21 +259,29 @@ static int load_db(void)
 }
 
 /*
- * Parse line on format "<name> <type> <IP-addr>".
+ * Parse line on format "<name> <type> <IP-addr> <ttl>".
  */
-static int parse_line (const char *str, int line, int *type_p, char **ip_p, u_int16 *port_p, char **name_p)
+static int parse_line(const char *str, int line, int *type_p, char **ip_p, u_int16 *port_p, char **name_p, unsigned int *ttl_p)
 {
    static char name[100+1];
    static char ip[MAX_ASCII_ADDR_LEN];
    static int port;
+   static u_int32 ttl;
    char type[10+1];
 
- DEBUG_MSG("%s:%d str '%s'", ETTER_DNS, line, str); 
+   DEBUG_MSG("%s:%d str '%s'", ETTER_DNS, line, str); 
 
-   if (sscanf(str,"%100s %10s %40[^\r\n# ]", name, type, ip) != 3) {
-      USER_MSG("dns_spoof: %s:%d Invalid entry %s\n", ETTER_DNS, line, str);
+   /* Set default TTL of 1 hour if not specified */
+   ttl = 3600;
+
+   if (sscanf(str,"%100s %10s %40[^\r\n# ] %u", name, type, ip, &ttl) < 3) { /* TTL should be optional - this should work */
+      USER_MSG("dns_spoof: %s:%d Invalid entry '%s'\n", ETTER_DNS, line, str);
       return (0);
    }
+   
+   if (ttl > 2147483647) ttl = 3600; /* keep TTL within DNS standard limits (2^31 - 1) - see RFC 2181 */
```

Comment:
for a better readability you should define MAX_DNS_TTL as 1 << 31 - 1 at the top using preprocessor and refer it in the if-statement.